src/function_data.jl

```diff
@@ -64,24 +64,49 @@ representation of cost functions where the points store quantities (x, y), such
 The curve starts at the first point given, not the origin.
 
 # Arguments
- - `points::Vector{Tuple{Float64, Float64}}`: the points (x, y) that define the function
+ - `points::Vector{@NamedTuple{x::Float64, y::Float64}}`: the points that define the function
 """
 struct PiecewiseLinearPointData <: FunctionData
-    points::Vector{Tuple{Float64, Float64}}
+    points::Vector{@NamedTuple{x::Float64, y::Float64}}
 
-    function PiecewiseLinearPointData(points)
+    function PiecewiseLinearPointData(points::Vector{<:NamedTuple{(:x, :y)}})
         _validate_piecewise_x(first.(points))
         return new(points)
     end
 end
 
+function PiecewiseLinearPointData(points::Vector{<:NamedTuple})
+    throw(
+        ArgumentError(
+            "If constructing PiecewiseLinearPointData with NamedTuples, points must have type Vector{<:NamedTuple{(:x, :y)}}; got $(typeof(points))",
+        ),
+    )
+end
+
+function _convert_to_xy(point)
```

Comment:
```suggestion
function _convert_to_named_tuple(point)
```

Reply:
How's `_convert_to_xy_coords` given the name of my new `NamedTuple` alias?